js/guestshare.js

```
@@ -79,12 +79,43 @@
 		},
 	};
 
+	var GuestShareConfig = {
+
+		load: function () {
+			return new Promise(function (resolve, reject) {
+				$.ajax({
+					type: "get",
+					url: OC.generateUrl('/apps/guests/shareconfig'),
+					success: function(data) {
+						resolve(data);
+					},
+					error: function(xhr, status) {
+						reject(Error(status));
+					}
+				});
+			});
+		}
+	}
+
 	OCA.Guests.GuestShare = GuestShare;
 
 	OCA.Guests.initGuestSharePlugin = function() {
 		OC.Plugins.register('OC.Share.ShareDialogView', {
 			attach: function (obj) {
 
+				// fetch guest share config, it is ok to get this async because
+				// if filtering wont be applied in frontend due to e.g. error it might
+				// be handled in the backend or entry not shown (depening on setting)
+				var guestShareConfig = null;
+				GuestShareConfig.load().then(
+					function (config) {
+						guestShareConfig = config;
```

I'm not sure about this. If the `guestShareConfig` isn't needed in the line 166, why risking having a different behavior?, and if it's needed, I don't think this guarantees a "valid" `guestShareConfig`. We could get the config too late.

---

we add config async to avoid blocking sharing just because some stupid config did no load due to error etc.

---

of course, the small parts of the code that need this config should throw an error etc.

---

```
obj.autocompleteHandler = function(search, response) {
  return oldHandler.call(obj, search, function(result, xhrResult) {
    GuestShareConfig.promise().then(function (config) {
      .....
      if (config && config.blockdomains) {
        for (i = 0 ; i < config.blockdomains.length; i++) {
          if (searchTerm.endsWith('@' + config.blockdomains[i])) {
            provideGuestEntry = false;
          }
        }
      } else {
      .....
    };
  };
}
```
I was thinking something like that. That way, we ensure that the config is always loaded or we show an error message.

---

@jvillafanez so generally, your suggestion would be to have the config loading being blocking for the guest sharing? meaning no config or invalid config = no guest sharing ?

---

If you need the config, you should guarantee the config is loaded. If the config isn't loaded because the request fails, or the config has wrong data, you can handle that error the way you want.

The scenario I'd want to avoid is that you get the config after you run the autocomplete handler, so the autocomplete could wrongly assume that there is no config when in fact it's just the request didn't finish.
This might not happen with the current code because the handler might be executed way too late, so the config has plenty of time to finish, but in general I think it's a bad idea.

You might want to add some caching in the config request instead.
Under normal circumstances, the code could run as I've proposed, so when you run the autocomplete handler, you make a request for the config and once that request finish you go through the guest code.
If you want to preload the config, you can make a request for the config early, assuming the config will be cached transparently, so the next request (the one that would be performed in the autocomplete handler) will use the cached config instead of requesting a new one.
In this case, we're guaranteeing that if there is a configuration, the autocomplete handler will run with it.